protocol/daemons/pricefeed/client/client.go

```diff
@@ -52,19 +52,25 @@ type Client struct {
 
 	// Ensure stop only executes one time.
 	stopDaemon sync.Once
+
+	// logger is the logger for the daemon.
+	logger log.Logger
 }
 
 // Ensure Client implements the HealthCheckable interface.
 var _ daemontypes.HealthCheckable = (*Client)(nil)
 
-func newClient() *Client {
+func newClient(logger log.Logger) *Client {
```

Comment:
update client to be created with a logger for the HealthCheckable initialization. Store it on the daemon for the `start` method.